lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,27 @@
+## Could Be A Palindrome
+
+# Write a method which takes a string as an argument and returns true if the letters could be re-arranged into a palindrome.
+
+# ```
+# palindrome_permutation?("hello") => false
+# palindrome_permutation?("carrace") => true # because racecar is a palindrome
+# ```
+
 
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
+  count = {}
+  x = 0
+    string.each_char do |char|
+      count[char]= string.count(char)
+    end
```

Comment:
`.count` has an O(n) loop, so this is an O(n^2) algorithm... Another place a hash could make this better.

```suggestion
    string.each_char do |char|
      count[char]= count[char] ? count[char] + 1 : 1
    end
```